src/shared/gitlab-client.ts

```diff
@@ -149,4 +159,77 @@ export class GitlabClient {
     };
   }
 
+  public async getUserByEmail(email: string) {
+    if (!validateEmail(email)) {
+      throw Boom.badRequest(`Invalid email '${email}'`);
+    }
+    const search = {
+      search: email,
+    };
+    const users = await this.fetchJson<User[]>(`users?${qs.stringify(search)}`, true);
+    if (!users || !users.length) {
+      throw Boom.badRequest(`Can\'t find user '${email}'`);
+    }
+    if (users.length > 1) {
+      // This shoud never happen
+      throw Boom.badRequest(`Found multiple users with email '${email}'`);
+    }
+    return users[0];
+  }
+
+  public createUser(
```

Comment:
OK, I can also accept this approach.

I think in we should in general be careful to not try to abstract GitLab itself away too much, as I'm afraid that could me an unnecessary abstraction that just makes things more complicated. Hiding some details of the API requests like this is not too much.